core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -75,6 +75,13 @@ private[spark] class CoarseMesosSchedulerBackend(
   val coresByTaskId = new HashMap[String, Int]
   var totalCoresAcquired = 0
 
+  // Ports acquired so far
+  // SlaveID ->  ports
+  val takenPortsPerSlave = new mutable.HashMap[String, List[Long]]
```

Comment:
Why keep track of this?  Mesos won't re-offer ports that have already been allocated?